code/game/gamemodes/cult/ghosts.dm

```diff
@@ -0,0 +1,236 @@
+
+	if(!bloodless)
+		var/list/choices = list()
+		for(var/obj/effect/decal/cleanable/blood/B in range(1))
+			if(B.amount > 0)
+				choices += B
+
+		if(!choices.len)
+			src << "<span class = 'warning'>There is no blood to use nearby.</span>"
+			return
+
+		var/obj/effect/decal/cleanable/blood/choice = input(src, "What blood would you like to use?") in null|choices
+		if(!choice)
+			return
+
+		if(choice.basecolor)
+			doodle_color = choice.basecolor
+
+	var/turf/simulated/T = get_turf(src)
+	if(!istype(T))
+		src << "<span class='warning'>You cannot doodle there.</span>"
+		return
+
+	var/num_doodles = 0
+	for(var/obj/effect/decal/cleanable/blood/writing/W in T)
+		num_doodles++
+	if(num_doodles > 4)
+		src << "<span class='warning'>There is no space to write on!</span>"
+		return
+
+	var/max_length = 50
+
+	var/message = sanitize(input("Write a message. It cannot be longer than [max_length] characters.", "Blood writing", ""))
+
+	if(message)
+		if(length(message) > max_length)
+			message += "-"
+			src << "<span class='warning'>You ran out of blood to write with!</span>"
+
+		var/obj/effect/decal/cleanable/blood/writing/W = new(T)
+		W.basecolor = doodle_color
+		W.update_icon()
+		W.message = message
+		W.add_hiddenprint(src)
+		if(!bloodless)
+			W.visible_message("<span class='warning'>Invisible fingers crudely paint something in blood on [T].</span>")
+		else
+			W.visible_message("<span class='warning'>Blood appears out of nowhere as invisible fingers crudely paint something on [T].</span>")
+
+	ghost_magic_cd = world.time + 30 SECONDS
+
+/mob/dead/observer/proc/shatter_glass()
+	set category = "Cult"
+	set name = "Noise: glass shatter"
+	set desc = "Make a sound of glass being shattered."
+
+	if(ghost_magic_cd > world.time)
+		src << "<span class='notice'>You need some more time before you can use your abilities.</span>"
+		return
+
+	playsound(loc, "shatter", 50, 1)
+
+	ghost_magic_cd = world.time + 10 SECONDS
+
+/mob/dead/observer/proc/slice()
+	set category = "Cult"
+	set name = "Noise: slice"
+	set desc = "Make a sound of a sword hit."
+
+	if(ghost_magic_cd > world.time)
+		src << "<span class='notice'>You need some more time before you can use your abilities.</span>"
+		return
+
+	playsound(loc, 'sound/weapons/bladeslice.ogg', 50, 1)
+
+	ghost_magic_cd = world.time + 10 SECONDS
+
+/mob/dead/observer/proc/move_item()
+	set category = "Cult"
+	set name = "Move item"
+	set desc = "Move a small item to where you are."
+
+	if(ghost_magic_cd > world.time)
+		src << "<span class='notice'>You need some more time before you can use your abilities.</span>"
+		return
+
+	var/turf/T = get_turf(src)
+
+	var/list/obj/item/choices = list()
+	for(var/obj/item/I in range(1))
+		if(I.w_class <= 2)
+			choices += I
+
+	if(!choices.len)
+		src << "<span class='warning'>There are no suitable items nearby.</span>"
+		return
+
+	var/obj/item/choice = input(src, "What item would you like to pull?") in null|choices
+	if(!choice || choice.w_class > 2)
```

Comment:
I know there probably aren't a whole lot of anchored /obj/item with a w_class of two or less, but could you add an anchored check just in-case?